libraries/AP_HAL_ChibiOS/CANFDIface.cpp

```diff
@@ -568,7 +568,7 @@ bool CANIface::init(const uint32_t bitrate, const uint32_t fdbitrate, const Oper
     if (can_ifaces[self_index_] == nullptr) {
         can_ifaces[self_index_] = this;
 #if !defined(HAL_BOOTLOADER_BUILD)
-        hal.can[self_index_] = this;
+        const_cast <AP_HAL::HAL&> (hal).can[self_index_] = this;
```

Comment:
This is kind of ugly.  Perhaps we should add a setter?

Reply:
I borrowed it from here: https://github.com/ArduPilot/ardupilot/blob/2e5af08a102d10b8b7d54414c7aa4f12f4a80ab9/libraries/AP_CANManager/AP_CANManager.cpp#L152

Reply:
It could be done by making a const setter and making the `can` array mutable. Should that be a separate commit on this PR or a separate PR?

Reply:
Fixed by making the `can` array `mutable`.